plugins/welcome/index.js

```diff
@@ -0,0 +1,49 @@
+const BasePlugin = require('../base');
+const Commenter = require('../../commenter');
+
+class WelcomePlugin extends BasePlugin {
+  /**
+   * Whether this plugin processes edit actions
+   * @public
+   * @override
+   * @returns {Boolean} Whether this plugin processes edit actions
+   */
+  get processesEdits() {
+    return false;
+  }
+
+  /**
+   * @typedef {import('@octokit/webhooks').WebhookPayloadPullRequest} WebhookPayloadPullRequest
+   * @typedef {WebhookPayloadPullRequest & { changes: Object }} WebhookPayloadPullRequestWithChanges
+   * @typedef {import('probot').Context<WebhookPayloadPullRequestWithChanges>} ProbotContext
+   */
+  /**
+   * Process a PR and check to see if we have a new friend
+   *
+   * @memberof WelcomePlugin
+   * @public
+   * @override
+   * @param {ProbotContext} context webhook context
+   * @param {Commenter} commenter Commenter
+   */
+  async processRequest(context, commenter) {
+    const message = process.env.WELCOME_MESSAGE || 'Thanks for making a contribution to the project!';
```

Comment:
You should follow the pattern used in the reviewers plugin instead:

- Pull in any default message in the plugin constructor (so you don't have to do this on every process request)
- If there is no message specified, perhaps just early return and don't even post a comment?
- Check the `config` parameter for an override message at the org or repo level (see https://github.com/godaddy/pullie/blob/master/plugins/reviewers/index.js#L68)

Reply:
updated based on feedback, added class property set in constructor from the .env var, am now bailing early if I don't have a message in either .env or in config, added checks for config passed from org or repo.